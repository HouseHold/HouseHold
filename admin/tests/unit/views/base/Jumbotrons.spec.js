import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Jumbotrons from '@/views/base/Jumbotrons'

Vue.use(CoreuiVue)

describe('Jumbotrons.vue', () => {
  it('has a name', () => {
    expect(Jumbotrons.name).toBe('Jumbotrons')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Jumbotrons)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Jumbotrons', () => {
    const wrapper = shallowMount(Jumbotrons)
    expect(wrapper.is(Jumbotrons)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Jumbotrons)
    expect(wrapper.element).toMatchSnapshot()
  })
})
