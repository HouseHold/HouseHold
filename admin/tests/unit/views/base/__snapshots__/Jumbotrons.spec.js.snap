// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotrons.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      tag="div"
    >
      <ccard-stub>
        <ccardheader-stub>
          <cicon-stub
            name="cil-justify-center"
          />
           
          <strong>
             Bootstrap Jumbotron 
          </strong>
           
          <div
            class="card-header-actions"
          >
            <a
              class="card-header-action"
              href="https://coreui.io/vue/docs/components/jumbotron"
              rel="noreferrer noopener"
              target="_blank"
            >
              <small
                class="text-muted"
              >
                docs
              </small>
            </a>
          </div>
        </ccardheader-stub>
         
        <ccardbody-stub>
          <cjumbotron-stub
            tag="div"
          >
            <h1
              class="display-3"
            >
              Bootstrap 4
            </h1>
             
            <p
              class="lead"
            >
              Bootstrap 4 Components for Vue.js 2.6+
            </p>
             
            <p>
              For more information visit website
            </p>
             
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              href="#"
              routertag="a"
              target="_self"
              type="button"
            >
              More Info
            </cbutton-stub>
          </cjumbotron-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
     
    <ccol-stub
      tag="div"
    >
      <ccard-stub>
        <ccardheader-stub>
          <cicon-stub
            name="cil-justify-center"
          />
           
          <strong>
             Jumbotron 
          </strong>
           
          <small>
            with slots
          </small>
        </ccardheader-stub>
         
        <ccardbody-stub>
          <cjumbotron-stub
            header="Bootstrap 4"
            lead=""
            tag="div"
          >
            <h1
              class="display-3"
            >
              Bootstrap 4
            </h1>
             
            <p
              class="lead"
            >
              
              This is a simple hero unit, a simple jumbotron-style component for
              calling extra attention to featured content or information.
            
            </p>
             
            <hr
              class="my-4"
            />
             
            <p>
              
              It uses utility classes for typography and spacing to space content
              out within the larger container.
            
            </p>
             
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              href="#"
              routertag="a"
              target="_self"
              type="button"
            >
              Do Something
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              color="success"
              event="click"
              exactactiveclass="active"
              href="#"
              routertag="a"
              target="_self"
              type="button"
            >
              Do Something Else
            </cbutton-stub>
          </cjumbotron-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      tag="div"
    >
      <ccard-stub>
        <ccardheader-stub>
          <cicon-stub
            name="cil-justify-center"
          />
           
          <strong>
             Jumbotron 
          </strong>
           
          <small>
            colors
          </small>
        </ccardheader-stub>
         
        <ccardbody-stub>
          <cjumbotron-stub
            bordercolor="dark"
            color="info"
            tag="div"
            textcolor="white"
          >
            <h1
              class="display-3"
            >
              Bootstrap 4
            </h1>
             
            <p
              class="lead"
            >
              
              This is a simple hero unit, a simple jumbotron-style component for
              calling extra attention to featured content or information.
            
            </p>
             
            <hr
              class="my-4"
            />
             
            <p>
              
              It uses utility classes for typography and spacing to space content
              out within the larger container.
            
            </p>
          </cjumbotron-stub>
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
</div>
`;
